internals/src/error.rs

```diff
@@ -26,3 +26,35 @@ macro_rules! write_err {
         }
     }
 }
+
+/// This module is used internally to check that error types implement certain traits.
+#[cfg(check_traits)]
+#[doc(hidden)]
```

Comment:
Maybe the module shouldn't be hidden but impls should be? Or at least macro should be visible, I think.

Reply:
Interesting, by adding
```
[package.metadata.docs.rs]
all-features = true
rustdoc-args = ["--cfg", "docsrs", "--cfg", "check_traits"]
```
and removing line 32 above, we get `check_pub_error` documented in the HTML at the crate root. And all the traits are only viewable if one goes into docs for `error` module and then `check_traits` module - seems like a good balance.

Reply:
Oh I didn't mention why I left the impls unhidden, looking at the HTML it is informative to see the traits because it shows really clearly what the macro does. I don't see any benefit in hiding them.